core/src/main/java/org/opensearch/sql/planner/physical/SessionContext.java

```diff
@@ -0,0 +1,23 @@
+package org.opensearch.sql.planner.physical;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+
+public interface SessionContext {
+  SessionContext None = new SessionContext() {
+    private final Clock epochStartClock = Clock.fixed(Instant.EPOCH, ZoneId.of("UTC"));
+    @Override
+    public Clock getQueryStartClock() {
+      return epochStartClock;
```

Comment:
I'd prefer if this throws a runtime exception since it shouldn't be used.

Reply:
That being said, it might be better to use Optionals and this default class has an empty clock defined.